hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/HttpServer2.java

```diff
@@ -649,6 +651,9 @@ private void initializeWebServer(Builder builder) throws IOException {
         if ((c instanceof AuthenticationFilterInitializer) && builder.securityEnabled) {
           addFilter("authentication",
               AuthenticationFilter.class.getName(), filterConfig);
+        } else if ((c instanceof ProxyUserAuthenticationFilterInitializer) && builder.securityEnabled) {
+          addFilter("authentication",
+              ProxyUserAuthenticationFilter.class.getName(), filterConfig);
```

Comment:
`ProxyUserAuthenticationFilterInitializer` adds the filter with a different name, not `"authentication"`.  It also injects some extra settings on top of the config obtained from `AuthenticationFilter#getFilterConfigMap`.  See https://github.com/apache/hadoop/blob/626b227094027ed08883af97a0734d2db7863864/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/authentication/server/ProxyUserAuthenticationFilterInitializer.java#L42-L58